File: backend/schemas/scene.py
from pydantic import BaseModel, Field


class ScriptRequest(BaseModel):
    script: str = Field(min_length=1)


from pydantic import BaseModel

class Scene(BaseModel):
    scene_number: int
    narration: str
    keyword: str
    visual_description: str
    image_prompt: str
    estimated_duration: int


class SceneAnalysisResponse(BaseModel):
    scenes: list[Scene]

File: frontend/src/types/scene.ts
export type ImageSearchResponse = {
  image_url: string;
  photographer: string;
  photographer_url: string;
  source: string;
};

export type Scene = {
  scene_number: number;
  narration: string;
  keyword: string;
  visual_description: string;
  image_prompt: string;
  estimated_duration: number;
  image?: ImageSearchResponse;
};

export type SceneAnalysisResponse = {
  scenes: Scene[];
};